pkg/util/bandwidth/interfaces.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bandwidth
+
+import "github.com/GoogleCloudPlatform/kubernetes/pkg/api/resource"
+
+type BandwidthShaper interface {
+	// Initialize bandwidth shaping for an interface (e.g. 'eth0', or 'cbr0')
+	InitializeInterface(iface string) error
```

Comment:
Purely from an API usability POV:

What happens if I don't call this?  What happens if I call it twice?  What happens if I call this after establishing some shaping rules?  Can't this be implied by other usage.  For example, I think a nicer interface would be something like:

```
shaper := bandwidth.NewShaper(eth0)
shaper.Limit(cidr, egress, ingress)
shaper.Reset(cidr)
shaper.ResetAll()
```